src/app.module.ts

```diff
@@ -1,18 +1,40 @@
-import { Module } from '@nestjs/common';
+import { Global, Module } from '@nestjs/common';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
 import { AuthModule } from './auth/auth.module';
 import { TypeOrmModule } from '@nestjs/typeorm';
 import { AppDataSource } from 'ormconfig';
 import { UserModule } from './user/user.module';
+import { ConfigService } from '@nestjs/config';
+import { ClientProxyFactory, Transport } from '@nestjs/microservices';
+import { configConstant } from './common/constants/config.constant';
 
+// Create rabbitMQ service to be used in other module
+const RabbitMQService = {
+  provide: 'IDENTITY_SERVICE',
+  useFactory: (configService: ConfigService) => {
+    return ClientProxyFactory.create({
+      transport: Transport.RMQ,
+      options: {
+        urls: [configService.get(configConstant.amq.url)],
+        queue: configService.get(configConstant.amq.queue),
+        queueOptions: {
+          durable: true,
+        },
+      },
+    })
+  },
+  inject: [ConfigService]
+}
+@Global()
 @Module({
   imports: [
     UserModule, 
```

Comment:
This is throwing error, you need to import ConfigModule.forRoot()

Reply:
sorry.. i have just fixed it..it is working now..

thanks..
u can pull again